datafusion/common/src/scalar.rs

```diff
@@ -1834,30 +1883,31 @@ impl ScalarValue {
                 }
 
                 let arr = Decimal128Array::from(vals)
-                    .with_precision_and_scale(*precision, *scale)
-                    .unwrap();
```

Comment:
Similar to previous [comment](https://github.com/apache/arrow-datafusion/pull/7901/files#r1381174929), this error is reachable because [ScalarValue::try_new_decimal128](https://github.com/apache/arrow-datafusion/blob/main/datafusion/common/src/scalar.rs#L660-L661) allows a decimal with precision 0, while [arrow-array](https://github.com/apache/arrow-rs/blob/master/arrow-array/src/types.rs#L1234-L1235) does not support that.
We can update `try_new_decimal128` to disallow decimal with precision 0 or establish an invariant here so this error becomes unreachable and we can panic here.